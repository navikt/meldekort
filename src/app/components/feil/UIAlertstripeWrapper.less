.alertstripe_wrapper {
  margin-top: 2rem;
}