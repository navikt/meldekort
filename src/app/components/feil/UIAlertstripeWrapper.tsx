import * as React from 'react';
import { BaksystemFeilmelding } from '../../types/ui';
import AlertStripe from 'nav-frontend-alertstriper';
import { RootState } from '../../store/configureStore';
import { selectFeilmelding } from '../../selectors/ui';
import { Dispatch } from 'redux';
import { UiActions } from '../../actions/ui';
import { connect } from 'react-redux';

interface MapStateToProps {
  baksystemFeilmelding: BaksystemFeilmelding;
}

interface MapDispatchToProps {
  skjulFeilmelding: () => void;
}

type UIAlertstripeWrapperProps = MapStateToProps & MapDispatchToProps;

const UIAlertstripeWrapper: React.FunctionComponent<UIAlertstripeWrapperProps> = ({ baksystemFeilmelding}) => { // { baksystemFeilmelding, skjulFeilmelding }
  return (
    <div className={'alertstripe_wrapper'}>
      {baksystemFeilmelding.visFeilmelding ? (
        <AlertStripe type="feil">
          <div>{baksystemFeilmelding.content()}</div>
        </AlertStripe>
      ) : null}
    </div>
  );
};

const mapStateToProps = (state: RootState): MapStateToProps => {
  return {
    baksystemFeilmelding: selectFeilmelding(state),
  };
};

const mapDispatchToProps = (dispatch: Dispatch): MapDispatchToProps => {
  return {
    skjulFeilmelding: () => dispatch(UiActions.skjulBaksystemFeilmelding()),
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(UIAlertstripeWrapper);
