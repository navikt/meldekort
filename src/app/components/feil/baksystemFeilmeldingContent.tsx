import * as React from 'react';
import { Undertittel } from 'nav-frontend-typografi';
import { FormattedMessage } from 'react-intl';

export const baksystemFeilmeldingContent = () => {
    return (
        <div>
            <Undertittel children={<FormattedMessage id="feilmelding.baksystem.overskrift"/>}/>
            <FormattedMessage id={'feilmelding.baksystem'} />
        </div>
    );
};