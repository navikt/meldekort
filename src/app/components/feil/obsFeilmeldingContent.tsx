import { Undertittel } from 'nav-frontend-typografi';
import { FormattedMessage } from 'react-intl';
import * as React from 'react';

export const obsFeilmeldingContent = () => {
    return (
        <div>
            <Undertittel children={<FormattedMessage id="obs.feilmelding.overskrift"/>}/>
            <FormattedMessage id={'obs.feilmelding'} />
        </div>
    );
};