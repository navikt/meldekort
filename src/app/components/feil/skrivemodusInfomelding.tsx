import React from 'react';
import AlertStripe from 'nav-frontend-alertstriper';
import { Skrivemodus } from '../../types/skrivemodus';
import { RootState } from '../../store/configureStore';
import { connect } from 'react-redux';
import { formatHtmlMessage, hentLocale } from '../../utils/intlUtil';

interface MapStateToProps {
  skrivemodus: Skrivemodus;
}

const SkrivemodusInfomelding: React.FunctionComponent<MapStateToProps> = props => {
  const skrivemodus = props.skrivemodus;

  const hentSkrivemodusInfomelding = () => {
    if (skrivemodus.melding === null) {
      return formatHtmlMessage("skrivemodusInfomelding")
    } else {
      return hentLocale() === 'nb'
        ? skrivemodus.melding.norsk
        : skrivemodus.melding.engelsk;
    }
  };

  return (
    <AlertStripe type={'info'}>{hentSkrivemodusInfomelding()}</AlertStripe>
  );
};

const mapStateToProps = (state: RootState): MapStateToProps => {
  return {
    skrivemodus: state.skrivemodus,
  };
};

export default connect<object, object, MapStateToProps>(mapStateToProps, null)(SkrivemodusInfomelding);
