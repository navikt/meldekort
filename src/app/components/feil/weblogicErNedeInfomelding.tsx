import React from 'react';
import AlertStripe from 'nav-frontend-alertstriper';
import { FormattedMessage } from 'react-intl';
import { WeblogicPing } from '../../types/weblogic';
import { RootState } from '../../store/configureStore';
import { connect } from 'react-redux';
import { hentLocale } from '../../utils/intlUtil';

interface MapStateToProps {
  weblogic: WeblogicPing;
}

const WeblogicErNedeInfomelding: React.FunctionComponent<MapStateToProps> = props => {
  let weblogic = props.weblogic;

  const hentNedetidsmelding = () => {
    if (weblogic.melding === null) {
      return <FormattedMessage id={'skrivemodusInfomelding'} />;
    } else {
      return hentLocale() === 'nb'
        ? weblogic.melding.norsk
        : weblogic.melding.engelsk;
    }
  };

  return <AlertStripe type={'info'}>{hentNedetidsmelding()}</AlertStripe>;
};

const mapStateToProps = (state: RootState): MapStateToProps => {
  return {
    weblogic: state.weblogic,
  };
};

export default connect(mapStateToProps, null)(WeblogicErNedeInfomelding);
