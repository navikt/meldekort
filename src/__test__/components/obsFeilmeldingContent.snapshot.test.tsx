import * as React from 'react';
import { obsFeilmeldingContent } from '../../app/components/feil/obsFeilmeldingContent';

it('ObsFeilmeldingContent', () => {
  expect(obsFeilmeldingContent()).toMatchSnapshot();
});
