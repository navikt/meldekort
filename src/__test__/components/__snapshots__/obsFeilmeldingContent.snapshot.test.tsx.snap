// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ObsFeilmeldingContent 1`] = `
<div>
  <Undertittel>
    <FormattedMessage
      id="obs.feilmelding.overskrift"
      values={Object {}}
    />
  </Undertittel>
  <FormattedMessage
    id="obs.feilmelding"
    values={Object {}}
  />
</div>
`;
